packages/driver/src/cypress/runner.ts

```diff
@@ -1602,30 +1602,37 @@ export default {
       },
 
       run (fn, response) {
-        Cypress.__actions = response?.actions
-
         if (_startTime == null) {
           _startTime = dayjs().toJSON()
         }
 
         _runnerListeners(_runner, Cypress, _emissions, getTestById, getTest, setTest, getTestFromHookOrFindTest)
 
-        return _runner.run((failures) => {
-          // if we happen to make it all the way through
-          // the run, then just set _runner.stopped to true here
-          _runner.stopped = true
-
-          // remove all the listeners
-          // so no more events fire
-          // since a test failure may 'leak' after a run completes
-          _runner.removeAllListeners()
-
-          // TODO this functions is not correctly
-          // synchronized with the 'end' event that
-          // we manage because of uncaught hook errors
-          if (fn) {
-            return fn(failures, getTestResults(_tests))
-          }
+        return Cypress.backend('get:run:actions')
```

Comment:
@MuazOthman Is there a reason we can't utilize the existing `get:cached:test:state` hook to pull this state before the run? We already request this before running to get both the run state and the test state.

This is [called in the event-manager](https://github.com/cypress-io/cypress/blob/feature/test-burn-in/packages/app/src/runner/event-manager.ts#L414) before starting the run: 

and is tied to [this existing backend action](https://github.com/cypress-io/cypress/blob/feature/test-burn-in/packages/server/lib/socket-base.ts#L496).


Also, I spend some time tracing this change through the features/test-burn-in branch, given this is sent from the Cloud, is there any reason we can't cache this on the server side when we get the cloud response before we start the tests all together without an additional backend request to start recorded tests?

Reply:
> is there any reason we can't cache this on the server side when we get the cloud response before we start the tests all together without an additional backend request to start recorded tests?
The data is needed to accessed synchronously within the patched `mocha` methods, so the solution was to load that data into the `Cypress` object. @brian-mann wants to refactor the approach subsequently (#28323)

Reply:
@emilyrohrbough I'm not sure if you're suggesting we make a change or asking for explanation.
If it is the former, can you elaborate on the change that should be made? if it is the latter, I don't really have deep enough knowledge to answer

Reply:
@MuazOthman Is it possible to remove this new backend request and leverage the existing `get:cached:test:state` for this data? This logic ends adding 2 additional client->server calls before Cypress even starts running.

Reply:
I'm not sure this is doable since these events happen at different stages due to data becoming available at different times. This new event is aimed at persisting/fetching actions received from the cloud and that happens well after the Cypress object is initialized.
@AtofStryker @brian-mann any ideas on this one?

Reply:
I'm not sure how to detect if we're recording, but I added logic to skip the server communication when running in interactive mode (`cypress open`). I would love to update the condition to cover all scenarios when we're not recording, suggestions welcome!